### PR TITLE
[polaris.shopify.com] Use `MDX` to render markdown

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

We want to leverage MDX to support richer content in our documentation site. MDX will allow us to easily create and use customized components to be used in `.md` files, and greatly simplifies our rendering and templating process.

### ⚠️ Disclaimer ⚠️ 
This is merging into the `next` branch in order to minimize merge conflicts with `v12` work

We want to merge in this PR in its current state in order to unblock documentation writing for component updates. In order to get things unblocked as quickly as possible, we are leaving some pages/builds broken intentionally while we complete the migration. 

Known issues include:

* parts of some patterns pages are broken because they are referencing the names of old layout components that are in the process of renaming and api changes
* some styling throughout the site may be a bit wonky while we migrate scss files and components

To see a list of minor issues please refer to the [project backlog](https://github.com/orgs/Shopify/projects/2250/views/118?type=beta)

### WHAT is this pull request doing?
* replacing `react-markdown` with `next-mdx-remote` for our markdown renderer
* removing most of the old `<!-- [component] -->` usages and swapping them out for mdx components
``` 
<!-- dodont -->

// becomes

<DoDont></DoDont>

// in .md files
```
* makes various content changes to `.md` files to make sure they are compatible with MDX
* Start to replace/rebuild/create some components using `polaris-react` layout primitives

### How to 🎩

1. Run the `dev` process for the website
2. Click through the website and make sure pages are working
3. Make note of the above known issues. If you discover another issue that isn't already documented please open an issue in the [backlog](https://github.com/orgs/Shopify/projects/2250/views/118?type=beta) and leave a comment in the PR

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: patches/playroom+0.28.0.patch [hunk 1]
```diff
@@ -218,6 +218,42 @@ index 9285b94..c268bdb 100644
...
+index 0183aaf..2d1d5d5 100644
+--- a/node_modules/playroom/src/Playroom/Preview.tsx
++++ b/node_modules/playroom/src/Playroom/Preview.tsx
+@@ -43,6 +43,16 @@ export default ({ themes, components, FrameComponent }: PreviewProps) => {
+ 
+   const resolvedTheme = themeName ? themes[themeName] : null;
+ 
++  // The SplashScreen is in place for security concerns to help mitigate
++  // phishing attacks (an attacker could add arbitrary code to the URL which is
++  // runin the target's browser tricking them into entering credentials, etc),
```
Comment (laurkim) [inline patches/playroom+0.28.0.patch]:
```suggestion
+  // run in the target's browser tricking them into entering credentials, etc),
```

File: polaris.shopify.com/content/components/deprecated/legacy-stack.md [hunk 2]
```diff
@@ -45,6 +42,26 @@ examples:
...
+
+<Lede>
+
+This is the legacy version of the Stack component which is in the process of being updated to a more flexible and composable API. Use to lay out a horizontal row of components or to achieve no-fuss vertical centering. A stack is made of flexible items that wrap each of the stack’s children. Options provide control of the wrapping, spacing, and relative size of the items in the stack.
+
+</Lede>
+
+<StatusBanner status={frontmatter.status}>
+  This component is no longer supported. The new [VerticalStack
+  component](/components/layout-and-structure/vertical-stack) can be used in
```
Comment (laurkim) [inline polaris.shopify.com/content/components/deprecated/legacy-stack.md]: VerticalStack was renamed to BlockStack on the `next` branch.

```suggestion
  This component is no longer supported. The new [BlockStack
  component](/components/layout-and-structure/block-stack) can be used in
```

File: polaris.shopify.com/content/components/deprecated/text-container.md [hunk 3]
```diff
@@ -26,6 +23,24 @@ examples:
...
+
+<Lede>
+
+A text container is used to wrap text elements such as paragraphs, headings, and lists to give them vertical spacing.
+
+</Lede>
+
+<StatusBanner status={frontmatter.status}>
+  The TextContainer is a simple layout component which was made redundant by the
+  VerticalStack component. Use VerticalStack with `gap` values of 2, 4, and 5 to
```
Comment (laurkim) [inline polaris.shopify.com/content/components/deprecated/text-container.md]: VerticalStack was renamed to BlockStack on the `next` branch.

```suggestion
  BlockStack component. Use BlockStack with `gap` values of 2, 4, and 5 to
```

**Review by laurkim — APPROVED** [review]
Nice! 🚀 🌟 

Looks great when testing locally 🙌  
Added some minor feedback but one thing I did notice is that with removing description from the component page metadata and using it directly in the `<Lede>`, we lose all the information on the `/components` root page. Imo it makes the page a bit inconsistent with the other ones that include descriptions on the grid (i.e., patterns page) so just curious if that's the intended final state for that page.

**Comment by yurm04** [comment]
@laurkim good catch on the components page! I think we'll have to figure out a fix forward for this, but it should not be the final stage for the site.

Created this ticket to tackle as soon as this gets merged in https://github.com/Shopify/polaris/issues/10364

**Review by kyledurand — APPROVED** [review]
👏

**Pull request merged**